app/jobs/conversations/detect_language_job.rb

```diff
@@ -0,0 +1,42 @@
+class Conversations::DetectLanguageJob < ApplicationJob
+  queue_as :default
+
+  def perform(message_id)
+    @message = Message.find(message_id)
+
+    return if hook.blank?
+
+    text = @message.content[0...1500]
+
+    response = client.detect_language(
+      content: text,
+      parent: "projects/#{hook.settings['project_id']}"
+    )
+
+    update_conversation(response)
+  end
+
+  private
+
+  def update_conversation(response)
+    conversation = @message.conversation
+    conversation_language = response.languages.first.language_code
+    additional_attributes = conversation.additional_attributes.merge({ conversation_language: conversation_language })
+
+    conversation.update!(additional_attributes: additional_attributes)
+
+    event = Events::Base.new('conversation_updated', Time.zone.now, { conversation: conversation })
+
+    AutomationRuleListener.instance.conversation_updated(event)
```

Comment:
@sojan-official I Had to add this manually as we don't support the 'conversation_updated' event in additional_attributes.

```
def notify_conversation_updation
    return unless previous_changes.keys.present? && (previous_changes.keys & %w[team_id assignee_id status snoozed_until custom_attributes label_list first_reply_created_at]).present?

    dispatcher_dispatch(CONVERSATION_UPDATED, previous_changes)
end
  ```

I didn't want to call it for all other listeners and grab the impact on this change.

Without this, it will work for other attributes on the conversation when you update something manually on the dashboard, not when you detect and update the language and then want to run the automation based on that. So for this case, I added the line.

Let me know wdyt?